### PR TITLE
Linux: add support for TextInputType.none

### PR DESCRIPTION
**Pull request by jpnurmi**
`TextInputType.none` makes it possible to disable the virtual keyboard for certain TextFields, and lays the foundations for custom in-app virtual keyboards (flutter/flutter#76072).

Ref: flutter/flutter#83567

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.
- [ ] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

**Comment by jpnurmi** [comment]
```dart
Column(
  children: const <Widget>[
    Padding(
      padding: EdgeInsets.all(24.0),
      child: TextField(
        keyboardType: TextInputType.none, // <==
        decoration: InputDecoration(hintText: 'TextInputType.none'),
      ),
    ),
    Padding(
      padding: EdgeInsets.all(24.0),
      child: TextField(
        keyboardType: TextInputType.text,
        decoration: InputDecoration(hintText: 'TextInputType.text'),
      ),
    ),
  ],
)
```
https://user-images.githubusercontent.com/140617/120719707-036fdd00-c4cb-11eb-8c15-264286bc0553.mp4

**Comment by jpnurmi** [comment]
Is there any reasonable way to test this?

File: shell/platform/linux/fl_text_input_plugin.cc [hunk 1]
```diff
@@ -58,6 +59,9 @@ struct FlTextInputPluginPrivate {
   // Send newline when multi-line and enter is pressed.
   gboolean input_multiline;
 
+  // The input method is not shown when the input type is "none".
+  gboolean input_none;
```
Comment (robert-ancell) [inline shell/platform/linux/fl_text_input_plugin.cc:63]: Can you replace `input_multiline`/`input_none` with a enum `input_type` - we can't be both none and multiline.

**Comment by robert-ancell** [comment]
> Is there any reasonable way to test this?

We don't currently have any test framework for testing at the GTK level.

**Review by robert-ancell — APPROVED** [review]
LGTM

**Review by cbracken — COMMENTED** [review]
Apologies for the delay -- thanks for adding!

**Pull request merged**